src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs

```diff
@@ -31,14 +38,161 @@ internal ModelOperationsCatalog(IHostEnvironment env)
+                if (dataLoader == null)
+                {
+                    // Try to see if the model was saved without a loader or a schema.
+                    if (ModelLoadContext.LoadModelOrNull<ITransformer, SignatureLoadModel>(_env, out var transformerChain, rep,
+                        CompositeDataLoader<object, ITransformer>.TransformerDirectory))
+                    {
+                        inputSchema = null;
+                        return transformerChain;
+                    }
+
+                    // Try to load from legacy model format.
+                    try
+                    {
+                        var loader = ModelFileUtils.LoadLoader(_env, rep, new MultiFileSource(null), false);
+                        inputSchema = loader.Schema;
+                        return TransformerChain.LoadFromLegacy(_env, stream);
+                    }
+                    catch (Exception ex)
+                    {
+                        throw _env.Except(ex, "Could not load legacy format model");
+                    }
+                }
+                if (dataLoader is CompositeDataLoader<IMultiStreamSource, ITransformer> composite)
+                {
+                    inputSchema = composite.Loader.GetOutputSchema();
+                    return composite.Transformer;
+                }
+                inputSchema = dataLoader.GetOutputSchema();
+                return new TransformerChain<ITransformer>();
+            }
+        }
+
+        /// <summary>
+        /// Load the model and its input schema from the stream.
+        /// </summary>
+        /// <param name="stream">A readable, seekable stream to load from.</param>
+        /// <returns>A model of type <see cref="CompositeDataLoader{IMultiStreamSource, ITransformer}"/> containing the loader
+        /// and the transformer chain.</returns>
+        public IDataLoader<IMultiStreamSource> Load(Stream stream)
+        {
+            _env.CheckValue(stream, nameof(stream));
+
+            using (var rep = RepositoryReader.Open(stream))
+            {
+                try
+                {
+                    ModelLoadContext.LoadModel<IDataLoader<IMultiStreamSource>, SignatureLoadModel>(_env, out var model, rep, null);
+                    return model;
+                }
+                catch (Exception ex)
+                {
+                    throw _env.Except(ex, "Model does not contain an IDataLoader");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Load a transformer model and a data loader model from the stream.
+        /// </summary>
+        /// <param name="stream">A readable, seekable stream to load from.</param>
+        /// <param name="loader">The data loader from the model stream.</param>
+        /// <returns>The transformer model from the model stream.</returns>
+        public ITransformer Load(Stream stream, out IDataLoader<IMultiStreamSource> loader)
```

Comment:
>Load [](start = 28, length = 4)

It would be nice if this was `LoadWithDataLoader`, so that we could have `mlContext.Model.Load(stream, out var thing)` without ambiguity.